README.md

```diff
@@ -253,6 +254,34 @@ If your desired ghc has been found, you use it to install haskell-language-serve
 ./cabal-hls-install data
 ```
 
+## HLS LSP Configuration
+
+haskell-language-server supports some forms of configuration.
+
+### Formatting providers
+By default, haskell-language-server is compiled with support for several different formatters.
+
+These include
+* `floskell`
+* `fourmolu`
+* `ormolu`
+* `stylish-haskell`
+* `brittany` (if compiled with AGPL)
+
+To choose one of them, you need to include the name of the one you want to use
+as the value for the `formattingProvider` key in your LSP configuration like so:
+```json
+{
+  ...
+  "haskell": {
+    ...
+    "formattingProvider": "fourmolu"
+    ...
+  }
+  ...
+}
```

Comment:
What is this lsp configuration? I have never seen that one.

Reply:
I'm assuming it is a `.json` file that you point your lsp client to (e.g. in `LanguageClient-neovim` that seems to be [g:LanguageClient_settingsPath](https://github.com/autozimu/LanguageClient-neovim/blob/next/doc/LanguageClient.txt#L221))

Reply:
Not sure if this is specific to `LanguageClient-neovim`? At least I have never seen such a configuration file for HLS itself. Others will have to clarify.

Reply:
I assumed it is generic, because I couldn't find another way of choosing a formatter. How do you do so in your editor+server?

Reply:
VSCode / VSCodium has a settings Tab where you can choose the formatter out of a dropbox menu

Reply:
For vscode under linux the file is located (e.g.) in `~/.config/Code`, and it contains the following things
```json
{
    "files.autoSave": "onFocusChange",
    "editor.tabSize": 2,
    "window.menuBarVisibility": "toggle",
    "window.zoomLevel": 0,
    "haskell.formattingProvider": "brittany",
    "haskell.hlintOn": false,
    "haskell.diagnosticsOnChange": false,
    "haskell.completionSnippetsOn": false
}
```
which includes the `haskell.formattingProvider` line.

Reply:
Ok makes sense, so this configuration's location is still editor dependent. We should make this clear and maybe refer to the documentation of the respective lsp-client.

Reply:
I pushed some "clarification", is it ok?

I'm also not sure how good an idea being this specific as I was is, because while it's helpful for someone who has never used their respective LSP client before, it might get outdated and is in a way "redundant" information.

Reply:
This is quite client-specific. For example, in `lsp-haskell` (the emacs mode), you configure these things with emacs `defcustom` variables. I think it's reasonable for the client to explain how configuration is done, since we can't specify it in a general way on the server side.

Reply:
But there should still be **some** mention that you can put such a field into your client's configuration, whatever that may be. If it didn't exist in the changelog you would have to read through source code to learn about this.

Besides, [there is already precedent](https://github.com/haskell/haskell-language-server#editor-integration) for descriptions on how to configure different clients for use with HLS.

Reply:
I don't think I'm the only one who found it took effort to find this out either:
https://twitter.com/seagreen__/status/1309967145221214209?s=20